proxy/http/HttpSM.cc

```diff
@@ -5310,9 +5310,6 @@ HttpSM::handle_post_failure()
     tunnel.deallocate_buffers();
     tunnel.reset();
     // Server died
-    vc_table.cleanup_entry(server_entry);
-    server_entry          = NULL;
-    server_session        = NULL;
```

Comment:
But at this point, we don't want to leave dangling pointers even if we do want to retry, right? If we retry we ought to get a new server session.